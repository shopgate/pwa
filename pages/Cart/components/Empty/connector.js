import { connect } from 'react-redux';
import goBackHistory from '@shopgate/pwa-common/actions/history/goBackHistory';

/**
 * Maps action dispatchers to the component props.
 * @param {function} dispatch The store dispatcher.
 * @return {Object} The extended component props.
 */
const mapDispatchToProps = dispatch => ({
  goBackHistory: () => dispatch(goBackHistory(1)),
});

export default connect(null, mapDispatchToProps);
