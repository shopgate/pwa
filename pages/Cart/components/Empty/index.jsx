import React from 'react';
import I18n from '@shopgate/pwa-common/components/I18n';
import Portal from '@shopgate/pwa-common/components/Portal';
import * as portals from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import Icon from './components/Icon';
import styles from './style';

/**
 * The Cart Empty component.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Empty = () => (
  <div className={styles.wrapper}>
    <Portal name={portals.CART_EMPTY_BEFORE} />
    <Portal name={portals.CART_EMPTY}>
      <div className={styles.container}>
        <div className={styles.icon}>
          <Icon />
        </div>
        <div className={styles.title} data-test-id="emptyCartPlaceHolderString">
          <I18n.Text string="cart.empty" />
        </div>
      </div>
    </Portal>
    <Portal name={portals.CART_EMPTY_AFTER} />
  </div>
);

export default Empty;
