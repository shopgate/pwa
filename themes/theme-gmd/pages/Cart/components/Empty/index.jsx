import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import Portal from '@shopgate/pwa-common/components/Portal';
import * as portals from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import RippleButton from '@shopgate/pwa-ui-shared/RippleButton';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';
import { encodeSVG } from '@shopgate/engage/core/helpers';
import Icon from './components/Icon';
import connect from './connector';
import styles from './style';

const { fullSVGs } = themeConfig;

/**
 * The Cart Empty component.
 * @param {Object} props The component props.
 * @return {JSX.Element}
 */
const Empty = ({ goBackHistory }) => {
  const { emptyCart } = fullSVGs;

  const imageSRC = useMemo(() => encodeSVG(emptyCart),
    [emptyCart]);

  return (
    <div className={styles.wrapper}>
      <Portal name={portals.CART_EMPTY_BEFORE} />
      <Portal name={portals.CART_EMPTY}>
        <div className={styles.container}>
          <div className={styles.icon}>
            {emptyCart ? <img src={imageSRC} alt="" /> : <Icon />}
          </div>
          <div className={styles.title} data-test-id="emptyCartPlaceHolderString">
            <I18n.Text string="cart.empty" />
          </div>
        </div>
      </Portal>
      <Portal name={portals.CART_EMPTY_AFTER} />
      <div className={styles.buttonContainer}>
        <RippleButton onClick={goBackHistory} className={styles.button} type="secondary">
          <I18n.Text string="cart.continue" />
        </RippleButton>
      </div>
    </div>
  );
};

Empty.propTypes = {
  goBackHistory: PropTypes.func.isRequired,
};

export default connect(Empty);
