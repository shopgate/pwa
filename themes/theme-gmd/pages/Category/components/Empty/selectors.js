import { createSelector } from 'reselect';
import {
  getCategoryProductCount,
  getCategoryChildCount,
} from '@shopgate/pwa-common-commerce/category/selectors';
import { getProductsResult } from '@shopgate/pwa-common-commerce/product/selectors/product';

/**
 * Determines whether or not a category has any content,
 * (subcategories, child products or products received via request).
 */
export const isCategoryEmpty = createSelector(
  getCategoryProductCount,
  getCategoryChildCount,
  getProductsResult,
  (productCount, childrenCount, { products, totalProductCount }) => {
    if (childrenCount !== 0) {
      return false;
    }

    if (productCount === 0) {
      return true;
    }

    if (totalProductCount !== null && products.length === 0) {
      return true;
    }

    return false;
  }
);
