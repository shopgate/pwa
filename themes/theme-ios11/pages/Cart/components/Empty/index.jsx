import React, { useMemo } from 'react';
import I18n from '@shopgate/pwa-common/components/I18n';
import Portal from '@shopgate/pwa-common/components/Portal';
import * as portals from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import { themeConfig } from '@shopgate/engage';
import { svgToDataUrl } from '@shopgate/engage/core/helpers';
import Icon from './components/Icon';
import styles from './style';

const { svgImages = {} } = themeConfig || {};

/**
 * The Cart Empty component.
 * @return {JSX.Element}
 */
const Empty = () => {
  const { emptyCart = '' } = svgImages || {};

  const imageSRC = useMemo(() => svgToDataUrl(emptyCart),
    [emptyCart]);

  return (
    <div className={styles.wrapper}>
      <Portal name={portals.CART_EMPTY_BEFORE} />
      <Portal name={portals.CART_EMPTY}>
        <div className={styles.container}>
          <div className={styles.icon}>
            {emptyCart ? <img src={imageSRC} alt="" /> : <Icon />}
          </div>
          <div className={styles.title} data-test-id="emptyCartPlaceHolderString">
            <I18n.Text string="cart.empty" />
          </div>
        </div>
      </Portal>
      <Portal name={portals.CART_EMPTY_AFTER} />
    </div>
  );
};

export default Empty;
