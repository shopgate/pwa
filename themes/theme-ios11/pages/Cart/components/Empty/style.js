import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const wrapper = css({
  display: 'flex',
  flexDirection: 'column',
  background: colors.light,
  textAlign: 'center',
  height: '100%',
}).toString();

const container = css({
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  justifyContent: 'center',
  flexGrow: '1',
  flexShrink: '0',
}).toString();

const icon = css({
  width: 216,
}).toString();

const title = css({
  textAlign: 'center',
  paddingTop: variables.gap.big * 2.25,
}).toString();

const buttonContainer = css({
  flexGrow: '0',
  padding: `${variables.emptyPage.buttonVerticalGap}px ${variables.gap.big}px`,
}).toString();

const button = css({
  width: '100%',
}).toString();

export default {
  wrapper,
  container,
  icon,
  title,
  buttonContainer,
  button,
};
