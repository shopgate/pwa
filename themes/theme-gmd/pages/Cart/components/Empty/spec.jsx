import React from 'react';
import { shallow } from 'enzyme';
import Empty from './index';

// Mock the redux connect() method instead of providing a fake store.
jest.mock('./connector', () => obj => obj);

describe('<CartEmpty />', () => {
  it('should render', () => {
    const wrapper = shallow(<Empty goBackHistory={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });
});
