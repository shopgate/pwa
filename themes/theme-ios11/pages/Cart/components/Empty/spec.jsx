import React from 'react';
import { shallow } from 'enzyme';
import Empty from './index';

describe('<CartEmpty />', () => {
  it('should render', () => {
    const wrapper = shallow(<Empty />);

    expect(wrapper).toMatchSnapshot();
  });
});
