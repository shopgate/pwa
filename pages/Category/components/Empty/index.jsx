import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Portal from '@shopgate/pwa-common/components/Portal';
import * as portals from '@shopgate/pwa-common/constants/Portals';
import NoResults from '@shopgate/pwa-ui-shared/NoResults';
import connect from './connector';
import { wrapper } from './style';

/**
 * The Empty component for the Category.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Empty = ({ isVisible, ...props }) => {
  if (!isVisible) {
    return null;
  }

  return (
    <Fragment>
      <Portal name={portals.NO_RESULTS_CONTENT_BEFORE} />
      <Portal name={portals.NO_RESULTS_CONTENT}>
        <NoResults {...props} className={wrapper} />
      </Portal>
      <Portal name={portals.NO_RESULTS_CONTENT_AFTER} />
    </Fragment>
  );
};

Empty.propTypes = {
  isVisible: PropTypes.bool.isRequired,
};

export default connect(Empty);
