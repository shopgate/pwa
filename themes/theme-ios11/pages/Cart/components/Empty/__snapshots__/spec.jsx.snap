// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartEmpty /> should render 1`] = `
<div
  className="css-glkfau"
>
  <Portal
    name="cart.empty.before"
    props={null}
  />
  <Portal
    name="cart.empty"
    props={null}
  >
    <div
      className="css-aphe06"
    >
      <div
        className="css-11342l"
      >
        <Icon />
      </div>
      <div
        className="css-b9wt33"
        data-test-id="emptyCartPlaceHolderString"
      >
        <Translate
          className=""
          params={Object {}}
          string="cart.empty"
        />
      </div>
    </div>
  </Portal>
  <Portal
    name="cart.empty.after"
    props={null}
  />
  <div
    className="css-1dq8tqy"
  >
    <RippleButton
      className="css-63oe3q"
      disabled={false}
      flat={false}
      onClick={[Function]}
      rippleClassName=""
      rippleSize={null}
      testId="Button"
      type="secondary"
      wrapContent={true}
    >
      <Translate
        className=""
        params={Object {}}
        string="cart.continue"
      />
    </RippleButton>
  </div>
</div>
`;
